src/stores/filter.js

```diff
@@ -37,12 +38,15 @@ export const useFilterStore = defineStore('filter', () => {
     if (!includeMature) {
       filterKeys = filterKeys.filter((filterKey) => filterKey !== 'mature')
     }
-    const mediaTypeFilters = {}
-    filterKeys.forEach((filterKey) => {
-      mediaTypeFilters[filterKey] = filters[filterKey]
-    })
-    return mediaTypeFilters
+    return filterKeys.reduce((obj, filterKey) => {
+      return { ...obj, [filterKey]: filters[filterKey] }
+    }, /** @type {Partial<import('../store/types').Filters>} */ ({}))
   }
+
+  const allFilterCategories =
+    /** @type {import('../store/types').FilterCategory[]} */ (
+      Object.keys(filters)
+    )
```

Comment:
It may be possible to avoid the cast by setting the type on allFilterCategories directly (though I'm not sure).